### PR TITLE
Add IValueProvider

### PR DESCRIPTION
**Pull request by frankbuckley**
Adds IValueProvider for observation values.

Identifies problems re unsigned integer types in value generator - will log issue.

### CODE REVIEW

File: gen/DSE.Open.Values.Generators/ValueTypesGenerator.cs [hunk 1]
```diff
@@ -567,6 +568,14 @@ a.NameEquals is not null
 
             ct.ThrowIfCancellationRequested();
 
+            var isUnsignedInteger = containedTypeSymbol.SpecialType is
+                SpecialType.System_Byte
+                or SpecialType.System_UInt16
+                or SpecialType.System_UInt32
+                or SpecialType.System_UInt64;
+
+            Debugger.Break();
```
Comment (rabuckley) [inline gen/DSE.Open.Values.Generators/ValueTypesGenerator.cs:577]: ```suggestion```

File: test/DSE.Open.Observations.Tests/MeasureTests.cs [hunk 2]
```diff
@@ -13,23 +12,23 @@ public sealed class MeasureTests
     [Fact]
     public void CanSerializeAndDeserialize_Binary()
     {
-        var measure = new Measure<bool>(s_measureUri, MeasurementLevel.Binary, "Test measure", "[subject] does something");
+        var measure = new Measure<Binary>(s_measureUri, MeasurementLevel.Binary, "Test measure", "[subject] does something");
         AssertJson.Roundtrip(measure);
```
Comment (rabuckley) [inline test/DSE.Open.Observations.Tests/MeasureTests.cs:16]: Not specific to this PR, but we can probably drop the reflection-based tests and rely on testing with the source generator.

**Review by rabuckley — APPROVED** [review]

File: benchmarks/DSE.Open.Benchmarks/Observations/ObservationHashValueBenchmarks.cs [hunk 3]
```diff
@@ -52,49 +51,49 @@ public ulong[] UsingGetRepeatableHashCode()
 
 public static class TestMeasures
 {
-    public static readonly Measure<bool> BinaryMeasure = new(
+    public static readonly Measure<Binary> BinaryMeasure = new(
```
Comment (rabuckley) [inline benchmarks/DSE.Open.Benchmarks/Observations/ObservationHashValueBenchmarks.cs:54]: Can we think of a name slightly more specific than just "Binary"?

Comment (frankbuckley) [reply benchmarks/DSE.Open.Benchmarks/Observations/ObservationHashValueBenchmarks.cs:54]: Not really

**Pull request merged**